ktor-server/ktor-server-core/jvm/src/io/ktor/routing/LocalPortRoutingBuilder.kt

```diff
@@ -19,11 +19,11 @@ import io.ktor.util.*
  *
  * @param port the port to match against
  *
- * @throws IllegalArgumentException if the port is outside the range of acceptable, non-ephemeral ports
+ * @throws IllegalArgumentException if the port is outside the range of TCP/UDP ports
  */
 @KtorExperimentalAPI
 fun Route.localPort(port: Int, build: Route.() -> Unit): Route {
-    require(port in 1..49151) { "Port $port must be a positive number within the range of non-ephemeral ports" }
+    require(port in 1..65535) { "Port $port must be a positive number between 1 and 65,535" }
```

Comment:
Entering bike shedding territory here but afaik 0 is also valid. A quick look in `InetSocketAddress` has:
```
private static int checkPort(int port) {
        if (port < 0 || port > 0xFFFF)
            throw new IllegalArgumentException("port out of range:" + port);
        return port;
    }
```

Reply:
According to [Wikipedia](https://en.wikipedia.org/wiki/List_of_TCP_and_UDP_port_numbers#Well-known_ports), `0` is reserved. It also adds that:
> in programming APIs (not in communication between hosts), [it] requests a system-allocated (dynamic) port

That wouldn't be useful as a filter.